### PR TITLE
feat(dropdown): ts conversion

### PR DESCRIPTION
**Pull request by EdwardIrby**
dropdown conversion

### CODE REVIEW

File: docs/src/content/components/dropdown.mdx [hunk 1]
```diff
@@ -288,7 +293,13 @@ import React from 'react'
 import Dropdown, { useDropdown } from '@pluralsight/ps-design-system-dropdown'
 import { CalendarIcon } from '@pluralsight/ps-design-system-icon'
 
-const DropdownWithIcon = React.forwardRef(
+interface DropdownWithIconProps extends Omit<HTMLAttributes<HTMLButtonElement>, 'onChange'> {
+  icon: React.ReactNode
+  onChange?: (e: React.MouseEvent, value: string) => void
```
Comment (danethurber) [inline docs/src/content/components/dropdown.mdx:298]: just double checking...should this be `value: ReactText`

File: packages/dropdown/package.json [hunk 2]
```diff
@@ -9,11 +9,12 @@
...
-    "build": "run-s build:js build:cjs build:css",
-    "build:cjs": "babel --root-mode upward src --out-dir dist/cjs",
+    "build": "run-s build:esm build:cjs build:css",
+    "build:cjs": "tsc --project tsconfig.build.json --module commonjs --target es5 --outDir dist/cjs",
     "build:css": "build-css --useGlamor -i dist/cjs/css/index.js",
-    "build:js": "cross-env ESM=true babel --root-mode upward src --out-dir dist/esm",
-    "build:watch": "yarn build:js -- --watch",
+    "build:esm": "tsc --project tsconfig.build.json --module es2015 --target es5 --outDir dist/esm",
+    "build:watch": "yarn build:esm -- --watch",
+    "prepublish": "yarn build",
```
Comment (danethurber) [inline packages/dropdown/package.json:17]:
```suggestion
```

i've made some changes and have gotten ride of the `prepublish` scripts

File: packages/dropdown/src/js/index.ts [hunk 3]
```diff
@@ -2,16 +2,40 @@ import {
...
-export const DropdownContext = createContext()
+import {
+  useCombinedRefs,
+  useMenuRef,
+  ValueOf
+} from '@pluralsight/ps-design-system-util'
+
+import * as vars from '../vars'
+
+export const DropdownContext = createContext('' as React.ReactText)
```
Comment (danethurber) [inline packages/dropdown/src/js/index.ts:21]:
```suggestion
export const DropdownContext = createContext<React.ReactText>('')
```

File: packages/dropdown/src/react/label.tsx [hunk 4]
```diff
@@ -13,12 +12,9 @@ const styles = ({ themeName }) => {
   )
 }
 
-export const Label = ({ label }) => {
+export const Label = ({ label }: { label?: React.ReactNode }) => {
```
Comment (danethurber) [inline packages/dropdown/src/react/label.tsx:15]:
```suggestion
export const Label: React.FC<{ label?: React.ReactNode }> = ({ label }) => {
```

File: packages/dropdown/src/react/sub-label.tsx [hunk 5]
```diff
@@ -13,12 +12,9 @@ const styles = ({ themeName }) => {
   )
 }
 
-export const SubLabel = ({ subLabel }) => {
+export const SubLabel = ({ subLabel }: { subLabel?: React.ReactNode }) => {
```
Comment (danethurber) [inline packages/dropdown/src/react/sub-label.tsx:15]:
```suggestion
export const SubLabel: React.FC<{ subLabel?: React.ReactNode }> = ({ subLabel }) => {
```

**Review by danethurber — APPROVED** [review]
looks great! couple minor changes 👍

File: docs/src/content/components/dropdown.mdx (hunk 1 above)
Comment (EdwardIrby) [reply docs/src/content/components/dropdown.mdx:298]: yep

**Pull request merged**